From 8b88fe089a089334d1be470a0283b49796b346cb Mon Sep 17 00:00:00 2001
From: Jessica Hamilton <jessica.l.hamilton@gmail.com>
Date: Sun, 18 Jul 2021 11:21:17 +0000
Subject: [PATCH 1/3] haiku_devel: use relative symlinks to libraries in haiku
 package.

* This helps with creating a sysroot, otherwise extracted package
  links refer to /system, which is outside of the sysroot.
* Should have no functional impact on Haiku, as these are already
  symlinks

Change-Id: I29f719dc2c0839dd090e7f33eea0b8f98e0d6355
---
 build/jam/packages/HaikuDevel          | 4 ++--
 build/jam/packages/HaikuDevelSecondary | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/build/jam/packages/HaikuDevel b/build/jam/packages/HaikuDevel
index f438f5539d9..bc7e69de34c 100644
--- a/build/jam/packages/HaikuDevel
+++ b/build/jam/packages/HaikuDevel
@@ -28,12 +28,12 @@ AddFilesToPackage lib : $(developmentLibs) ;
 # library symlinks
 local lib ;
 for lib in [ HaikuImageGetSystemLibs ] $(developmentLibs) {
-	AddSymlinkToPackage develop lib : /system/lib $(lib:BS) ;
+	AddSymlinkToPackage develop lib : ../../lib $(lib:BS) ;
 	local abiVersion = [ on $(lib) return $(HAIKU_LIB_ABI_VERSION) ] ;
 	if $(abiVersion) {
 		local abiVersionedLib = $(lib:BS).$(abiVersion) ;
 		AddSymlinkToPackage develop lib
-			: /system/lib $(abiVersionedLib) ;
+			: ../../lib $(abiVersionedLib) ;
 	}
 }
 
diff --git a/build/jam/packages/HaikuDevelSecondary b/build/jam/packages/HaikuDevelSecondary
index 6c1f36b61fc..ae7ef4c8887 100644
--- a/build/jam/packages/HaikuDevelSecondary
+++ b/build/jam/packages/HaikuDevelSecondary
@@ -23,12 +23,12 @@ AddFilesToPackage lib $(architecture) : $(developmentLibs) ;
 local lib ;
 for lib in [ HaikuImageGetSystemLibs ] $(developmentLibs) {
 	AddSymlinkToPackage develop lib $(architecture)
-		: /system/lib/$(architecture) $(lib:BS) ;
+		: ../../lib/$(architecture) $(lib:BS) ;
 	local abiVersion = [ on $(lib) return $(HAIKU_LIB_ABI_VERSION) ] ;
 	if $(abiVersion) {
 		local abiVersionedLib = $(lib:BS).$(abiVersion) ;
 		AddSymlinkToPackage develop lib $(architecture)
-			: /system/lib/$(architecture) $(abiVersionedLib) ;
+			: ../../lib/$(architecture) $(abiVersionedLib) ;
 	}
 }
 
-- 
2.25.1

